lxd/daemon.go

```diff
@@ -399,7 +399,23 @@ func (d *Daemon) SetupStorageDriver() error {
 	return err
 }
 
+// have we setup shared mounts?
+var sharedMounted bool
+var sharedMountsLock sync.Mutex
+
 func setupSharedMounts() error {
+
+	if sharedMounted {
+		return nil
+	}
+
+	sharedMountsLock.Lock()
+	defer sharedMountsLock.Unlock()
+
+	if sharedMounted {
+		return nil
+	}
```

Comment:
Quoting Stéphane Graber (notifications@github.com):
> >  func setupSharedMounts() error {
> > +
> > +	if sharedMounted {
> > +		return nil
> > +	}
> > +
> > +	sharedMountsLock.Lock()
> > +	defer sharedMountsLock.Unlock()
> > +
> > +	if sharedMounted {
> > +		return nil
> > +	}
> 
> Hmm, why the same check before and after acquiring the lock?

...

before to avoid lock overhead in most cases.

after bc before is meaningless if it was false.